Readme.md

```diff
@@ -1,18 +1,44 @@
-# gdal
+# GDAL
 
 [![Documentation](https://docs.rs/gdal/badge.svg)](https://docs.rs/gdal)
 ![Build Status](https://github.com/georust/gdal/workflows/CI/badge.svg)
 
-[GDAL](http://gdal.org/) bindings for [Rust](http://www.rust-lang.org/).
+[GDAL](http://gdal.org/) is a translator and processing library for various raster and vector geospatial data formats. 
 
-So far, you can:
+This crate provides safe, idiomatic [Rust](http://www.rust-lang.org/) bindings for GDAL.
 
-* open a raster dataset for reading/writing
-* get size and number of bands
-* get/set projection and geo-transform
-* read and write raster data
-* warp between datasets
-* read and write vector data
-* access metadata
+## Capabilities
 
-Many raster and vector functions are not available. Patches welcome :)
+GDAL is an incredibly powerful library. For a general understanding of its capabilities, a good place to get started is the [GDAL User-oriented documentation](https://gdal.org/user/index.html). These features include:
+
+* Open a raster dataset for reading/writing
+* Get the raster size and number of bands
+* Get and set the projection and geo-transform
+* Warp (resample and reproject) between datasets
+* Read and write vector data
+* Read and write metadata
+* Translate between formats
+
+## Documentation
+
+This crate's [API documentation](https://docs.rs/crate/gdal) is hosted on [docs.rs](https://docs.rs). The Rust documentation is currently a work in progress, and may not cover requisite details on parameter semantics, value interpretation, etc.. Therefore, the authoritative documentation for the [GDAL C API](https://gdal.org/api/index.html#c-api) may be a helpful reference.
```

Comment:
It might be worth linking to the C++ API too, since it's matches the Rust one better and usually has more details.

Reply:
@lnicola I made this point about C vs C++ because when I was new to the crate, I found it confusing on where to go for semantic details, especially when some of the names are the same or similar. I assumed we only link to C because that's what `bindgen` provides for us, and then there's larger complexities with name-mangling, memory model, libraries, etc. in C++. 

Were you thinking I should strike this section, or refine it, or was this just a general point?

Reply:
Of course, we're calling the C API, but ours is more similar in spirit with the C++ one. Anyway, my point was about the documentation: the C GDAL docs are pretty bare, while the C++ ones have more details.

Compare https://gdal.org/doxygen/gdal_8h.html#a764cb93a266ada902b13c2bdad3f7037 with https://gdal.org/doxygen/classGDALDataset.html#ad2a643880bf8b3c492783e1ab5fbc667.

So I was thinking we could put it more like:

```
Therefore, the authoritative documentation is that of GDAL in the form of its [C](https://gdal.org/api/index.html#c-api) and [C++](https://gdal.org/api/index.html#id3) APIs.
The latter is usually better documented.
```

Reply:
I think I misread your comment... more clear now that I came back to it. My point is somewhat relevant still; when I was just coming to this crate, I would try to translate from the C++ docs to Rust, and hit a wall because we don't implement the C++ functionality, and didn't to save folks from that pothole.

Reply:
> because we don't implement the C++ functionality

Sure, but our API doesn't look like the C one either :smile:.